pkg/template/jsonpath.go

```diff
@@ -0,0 +1,155 @@
+package template
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"regexp"
+
+	"k8s.io/client-go/third_party/forked/golang/template"
+	"k8s.io/client-go/util/jsonpath"
+)
+
+var (
+	tektonVar  = regexp.MustCompile(`^\$\((.*)\)$`)
+	jsonRegexp = regexp.MustCompile(`^\{\.?([^{}]+)\}$|^\.?([^{}]+)$`)
+)
+
+func ParseJSONPath(input interface{}, expr string) (string, error) {
```

Comment:
Add godoc comments to describe the inputs/outputs

Reply:
done